Following Ansible Best Practices 2.

nhosoi opened the pull request:
- logging_collector -> logging_provider ($ROLENAME_provider).
- In rules.conf.j2, replacing double quotes with single quotes.
- Renamed <filename>.yaml to <filename>.yml

nhosoi committed:
Following Ansible Best Practices 2.

- logging_collector -> logging_provider ($ROLENAME_provider).
- In rules.conf.j2, replacing double quotes with single quotes.
- Renamed <filename>.yaml to <filename>.yml
- Changing rsyslog_elasticsearch to the internal variable __rsyslog_elasticsearch
  which is set via logging_outputs configured by users.
- Changing rsyslog_enabled to the internal variable __rsyslog_enabled which is
  set via logging_enabled configured by users.
- Adding and fixing comments and docs.

pcahyna commented on tasks/main.yml:
```diff
@@ -1,7 +1,6 @@
+  block:
 
     - name: Set rsyslog_enabled
       set_fact:
```
given that everything is in a block, could you set this in a variable on the block, instead of using set_fact? Or are you using this fact even outside the block?

nhosoi replied:
If we include_role "{{ role_path }}/roles/rsyslog" in the block, then can we say everything under roles/rsyslog is in the block?  If the answer is yes, then we could set them in a variable on the block.

pcahyna replied:
Should be the case, you are using this already in another PR, no?

nhosoi replied:
Well, these set_fact calls were untouched in https://github.com/linux-system-roles/logging/pull/87.  The use case of the variables here are quite different from the other cases.  The vars fixed in pr/87 are used in the included role once, immediately.  But the vars set in this block by set_fact are supposed to be global and used in many places some times in the nested way...

nhosoi committed:
- Removing unused rule variables rsyslog_rules, rsyslog_group_rules,
  rsyslog_host_rules and rsyslog_dependent_rules from
  roles/rsyslog/defaults/main.yaml.
- Fixing obsolete comments.

nhosoi commented:
Closing in favour of https://github.com/nhosoi/logging/pull/3.
